src/GitHub.Exports.Reactive/Collections/TrackingCollection.cs

```diff
@@ -0,0 +1,842 @@
+                        idx = ~idx;
+                    if (idx == list.Count)
+                        ret = new ActionData(TheAction.Add, item, null, list.Count, -1, list);
+                    else
+                        ret = new ActionData(TheAction.Insert, item, null, idx, -1, list);
+                }
+            }
+            else
+                ret = new ActionData(TheAction.Add, item, null, list.Count, -1, list);
+            return ret;
+        }
+
+        ActionData SortedNone(ActionData data)
+        {
+            if (data.TheAction != TheAction.None)
+                return data;
+            data.List[data.OldPosition].CopyFrom(data.Item);
+            return data;
+        }
+
+        ActionData SortedAdd(ActionData data)
+        {
+            if (data.TheAction != TheAction.Add)
+                return data;
+            data.List.Add(data.Item);
+            return data;
+        }
+
+        ActionData SortedInsert(ActionData data)
+        {
+            if (data.TheAction != TheAction.Insert)
+                return data;
+            data.List.Insert(data.Position, data.Item);
+            UpdateIndexCache(data.Position, data.List.Count, data.List, sortedIndexCache);
+            return data;
+        }
+        ActionData SortedMove(ActionData data)
+        {
+            if (data.TheAction != TheAction.Move)
+                return data;
+            data.OldItem.CopyFrom(data.Item);
+            var pos = FindNewPositionForItem(data.OldPosition, data.Position < 0, data.List, comparer, sortedIndexCache);
+            // the old item is the one moving around
+            return new ActionData(data.TheAction, data.OldItem, null, pos, data.OldPosition, data.List);
+        }
+
+        ActionData FilteredAdd(ActionData data)
+        {
+            if (data.TheAction != TheAction.Add)
+                return data;
+
+            if (data.IsIncluded)
+                InternalAddItem(data.Item);
+            return data;
+        }
+
+        ActionData CalculateIndexes(ActionData data)
+        {
+            data.Index = GetIndexFiltered(data.Item);
+            data.IndexPivot = GetLiveListPivot(data.Position, data.List);
+            return data;
+        }
+
+        ActionData FilteredNone(ActionData data)
+        {
+            if (data.TheAction != TheAction.None)
+                return data;
+
+            // nothing has changed as far as the live list is concerned
+            if ((data.IsIncluded && data.Index >= 0) || !data.IsIncluded && data.Index < 0)
+                return data;
+
+            // wasn't on the live list, but it is now
+            if (data.IsIncluded && data.Index < 0)
+                InsertAndRecalculate(data.List, data.Item, data.IndexPivot, data.Position, false);
+
+            // was on the live list, it's not anymore
+            else if (!data.IsIncluded && data.Index >= 0)
+                RemoveAndRecalculate(data.List, data.Item, data.Index, data.Position);
```

Comment:
The unit tests don't cover this code path. There should probably be a test for this case.